### PR TITLE
AB#703 elaborate description of remote attestation

### PR DESCRIPTION
**Pull request by katexochen**

### CODE REVIEW

File: content/docs/features/attestation.md [hunk 1]
```diff
@@ -7,18 +7,53 @@ weight: 1
 
 # Attestation
 
-Hardware-rooted *remote attestation* is a key ingredient for distributed confidential apps. Without it, services couldn't trust each other and clients couldn't trust the app. Thus, Marblerun relies heavily on the *Data Center Attestation Primitives* (DCAP) of the latest SGX-enabled Intel Xeon processors. You can learn more about DCAP in the [official Intel DCAP orientation](https://download.01.org/intel-sgx/sgx-dcap/1.9/linux/docs/Intel_SGX_DCAP_ECDSA_Orientation.pdf).
+Hardware-rooted *remote attestation* is a key ingredient for distributed confidential apps. Marblerun relies heavily on the [*Data Center Attestation Primitives* (DCAP)](https://download.01.org/intel-sgx/sgx-dcap/1.9/linux/docs/Intel_SGX_DCAP_ECDSA_Orientation.pdf) of the latest SGX-enabled Intel Xeon processors.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:10]: Why cutting the introduction, too wordy?

File: content/docs/features/attestation.md [hunk 2]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
 
-## Internal attestation of Marbles
+## Coordinator deployment
 
-The Coordinator verifies the identity and integrity of each newly spawned Marble before admitting it to the mesh.
-A freshly spawned Marble will try to register and activate itself by sending an activation request via gRPC to the Coordinator.
-This request contains a remote attestation certificate (or "quote"), which allows the Coordinator to verify that the service adheres to the Manifest in effect.
+In a first step, the Marblerun Coordinator is deployed to the cluster.
+The Coordinator generates a X.509 certificate chain with a root and intermediate certificate authority (CA).
+It generates a remote attestation quote which contains the hash of the root certificate as well as a hardware measurement of the enclave the Coordinator is running in.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:18]:
```suggestion
Initially, the Marblerun Coordinator is deployed to the cluster.
The Coordinator generates an X.509 certificate chain with a root and intermediate certificate authority (CA).
It generates a remote attestation quote which contains the hash of the root certificate as well as a hardware measurement of its enclave.
```

File: content/docs/features/attestation.md [hunk 3]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
 
-## External attestation of the app
+The admin revices the quote and verifies that measurement matches the Coordinator's binary to verify there was no manipulation to the Coordinators code.
+The admin also revices the certificate chain of the Coordinator and verifies that the hash of the root certificate is contained in the quote.
+This ensures that the certificate was indeed generated by the Coordinator and wasn't manipulated during transport.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:22]:
```suggestion
The admin receives the quote and verifies the Coordinator's integrity via the measurement. Next, they verify that the quote contains the root certificate’s hash. This ensures that the certificate was indeed generated by this Coordinator and was not manipulated during transport.
```

File: content/docs/features/attestation.md [hunk 4]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
 
-To keep things simple, the Coordinator issues one concise remote-attestation statement for your whole distributed app. Technically, the statement resembles a regular TLS certificate chain with some extra info. (TLS is the protocol that is responsible for the lock in your browser's toolbar ;-). The chain ends at the root certificate authority (CA) of the provider of the secure hardware, which currently in almost all cases will be Intel.
+After the successful verification of the Coordinator, the manifest can be uploaded to the cluster.
+The manifest describes which Marbles (services) should be deployed to your cluster. It contains a measurement values of the Marbles which were made after the Marbles were built.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:25]:
```suggestion
The manifest describes which Marbles (services) should be deployed to your cluster. It contains their enclaves’ build-time measurements. The manifest is described in more detail in our [“defining a manifest”] ]({{< ref " docs/workflows/define-manifest.md">}}) section. 
```

File: content/docs/features/attestation.md [hunk 5]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
+![Coordinator deployment](/img/coordinator_deployment.svg)
+
+## Marble deployment and attestation
+
+The Coordinator enforces the manifest and deploys the desired Marble(s) to the cluster.
+Each Marble registers itself with a quote via gRPC at the coordinator.
+The quote and the manifest both contain measurement of the Marble, and the Coordinator compares both to ensure the Marble's identity and integrity.
+The [Marble's TLS credentials]({{< ref "docs/features/secrets-management.md" >}}) are generated by the coordinator and signed with the intermediate CA certificate.
+The Coordinator's certificate Chain was well as the freshly generated Marble certificate are then delivered to the Marble.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:35]:
```suggestion
This concludes the Coordinator’s initialization. From then on it enforces the manifest and only admits the desired Marble(s) to the cluster.
Each Marble registers itself with a quote via gRPC to the Coordinator.
The quote contains this Marble's measurements. The Coordinator compares them to the manifest to ensure the Marble's identity and integrity.
The [Marble's TLS credentials]({{< ref "docs/features/secrets-management.md" >}}) are generated by the Coordinator and signed with the intermediate CA certificate.
The Coordinator's certificate chain as well as the freshly generated Marble certificate are then delivered to the Marble.
```

File: content/docs/features/attestation.md [hunk 6]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
+
+![Marble deployment](/img/marble_deployment.svg)
+
+## Confidential service mesh
+
+The Marble and the Coordinator can now communicate securely and authenticated over TLS (with client authentication), using the Coordinator's CA a root of trust.
+Different Marbles can communicate with each other in the same way.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:42]:
```suggestion
The Marble and the Coordinator can now communicate securely and authenticated over TLS (with client authentication), using the Coordinator's CA as a root of trust.
Different Marbles can communicate with each other in the same way.
```

File: content/docs/features/attestation.md [hunk 7]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
+
+If a client (i.e. a relying party, a customer) wants to connect to a Marble, the Coordinators certificate is used to establish a TLS connection and verify the Marble's certificate.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:44]:
```suggestion
If a client (i.e., a relying party, a customer) wants to connect to a Marble, the Coordinator’s certificate is used as a trusted CA to establish a TLS connection and verify the Marble's certificate.
```

File: content/docs/features/attestation.md [hunk 8]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
+
+![service mesh](/img/service_mesh.svg)
+
+## Cluster attestation
+
+The Coordinator issues one concise remote-attestation statement for your whole distributed app.
+A measurement of the coordinator is generated after the Coordinator's build and distributed to the relying party (this has to be done by the admin or operator).
+The relying party requests the Coordinator's quote and certificate chain.
+The quote contains the hash of the Coordinator's root CA certificate, with which the received certificate chain is verified.
+The quote also contains a measurement of the Coordinator, which is verified against the measurement received by the admin or operator.
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:54]:
```suggestion
The Coordinator issues one concise remote-attestation statement for your whole distributed app.
The Coordinator's build-time measurement is distributed to the relying party (this must be done by the admin or operator).
The relying party requests the Coordinator's quote and certificate chain.
The quote contains the hash of the Coordinator's root CA certificate, which is verified against the received certificate chain.
The quote also contains the Coordinator's measurement, which is verified against the build-time measurement.
```

File: content/docs/features/attestation.md [hunk 9]
```diff
@@ -7,18 +7,53 @@ weight: 1
...
+
+The relying party then request the manifest from the Coordinator and ensures it contains the expected Marbles and the expected measurement values of those Marbles.
+The steps required on the client-side are described in our [verification hands-on]({{< ref "docs/workflows/verification.md" >}}).
```
Comment (m1ghtym0) [inline content/docs/features/attestation.md:57]:
```suggestion
The relying party then requests the manifest from the Coordinator and ensures it contains the expected Marbles and their expected measurements.
The steps required on the client-side are described in our [verification hands-on]({{< ref "docs/workflows/verification.md" >}}).
```

**Review by m1ghtym0 — CHANGES_REQUESTED** [review]
Thanks Paul!
I like those graphics. We should make sure we use one consistent design principle for graphics through the docs.
I'm not really design-savvy and would like to hear @flxflx opinion on the design?
Either way, we should document how we create those designs in our wiki for future reference.

File: content/docs/features/attestation.md (hunk 1 above)
Comment (katexochen) [reply content/docs/features/attestation.md:10]: yes, I wanted to cap the total length of the article.

File: content/docs/features/attestation.md (hunk 3 above)
Comment (katexochen) [reply content/docs/features/attestation.md:22]: Less linebreaks in the code?

Comment (m1ghtym0) [reply content/docs/features/attestation.md:22]: Removing the linebreaks was unintentional, but I tried to shorten this section a bit.

**Review by flxflx — APPROVED** [review]
Graphics look great - thank you!

**Review by m1ghtym0 — APPROVED** [review]
